release notes/v0.43.0.md

```diff
@@ -0,0 +1,124 @@
+k6 v0.43.0 is here! :tada:
+
+Notable changes in this release include:
+- [xk6-browser](https://github.com/grafana/xk6-browser) is now bundled in k6 as an experimental module, and usable without a separate binary or compilation step!
+- Native support for JavaScript's `async`/`await`.
+- A new experimental JavaScript module for distributed tracing.
```

Comment:
```suggestion
- A new experimental module for distributed tracing.
```

It is implemented in go at this point

Reply:
Well, sure, as all built-in k6 JS modules are, but from the perspective of the user, it's still a JS module.

Reply:
I prefer to not call them `JS modules` as I prefer this name for actually what is known in the specification as "Source text modules" (which is a great name). 

Our in go implemented modules are mostly called (by me at least) "go modules" which isn't great given that there is such a thing as "go modules" in go ;). 

But I would also argue that you are *not* mentioning this for the browser above - which is also an experimental (JavaScript/go) module

Reply:
Alright, but the ES spec surely wouldn't refer to them as "JS modules" either, as "JS" is an implementation, so they use an internal name. For our users, however, "experimental JS module" gives a lot more detail than just "experimental module", so I'd prefer keeping it.

Not mentioning it for the browser was an oversight, and looking back, I do prefer "xk6-browser is now a built-in JavaScript module", so I can change it.

The reason we should clarify this is that we have different types of extensions: outputs and JS modules. Calling them just "modules" lacks the relevant and IMO useful context.

I don't want to split hairs about semantics here, so let's ask for a tie-breaking opinion. Can someone else chime in? :pray:

Reply:
We do use "k6 modules natively supported" in the [docs](https://k6.io/docs/javascript-api/)  and [built-in modules](https://k6.io/docs/using-k6/modules/#built-in-modules) in another place.

Reply:
Alright, you convinced me. Will stop using "JS module" from now on then. :handshake: